src/_pytest/terminal.py

```diff
@@ -879,16 +881,13 @@ def mkrel(nodeid):
             if self.verbosity >= 2 and nodeid.split("::")[0] != fspath.replace(
                 "\\", nodes.SEP
             ):
-                res += " <- " + bestrelpath(self.startpath, fspath)
+                res += " <- " + bestrelpath(self.startpath, Path(fspath))
         else:
             res = "[location]"
         return res + " "
 
-    def _getfailureheadline(self, rep):
-        head_line = rep.head_line
```

Comment:
Looking again, this code looks like it was written this way so that the `head_line` property is only evaluated once. Probably doesn't matter, but I'll keep it this way.